src/Chapter2.hs

```diff
@@ -857,7 +872,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind = go []
+  where
+    go :: [a] -> [a] -> [a]
+    go res [] = res
+    go res xs = foldl (flip (:)) res xs
```

Comment:
Wow, that's a clever solution using standard functions! :exploding_head: 

Btw, you don't need to handle the case with `[]` in the `go` function, since `foldl` already takes care of this. And, after you do this, you can notice, that there is no need in the `go` function with such implementation, and you implement `rewind` in a single line :slightly_smiling_face:

Reply:
Thats sounds nice!